### PR TITLE
Yes / no control

### PR DESCRIPTION
**Pull request by pdolega**
Working control here: http://mestudent.github.io/react-formalin/sample/page/sample.html

### CODE REVIEW

File: sample/SamplePage.tsx [hunk 1]
```diff
@@ -3,18 +3,14 @@ import * as ReactDOM from "react-dom";
 import {SamplePageTextInput} from "./SamplePageTextInput";
```
Comment (pdolega) [inline sample/SamplePage.tsx]: wrote custom router here - it's only couple of lines of code and I didn't want to pull another dependency (react-router) only to have it on sample page

File: src/base/Input.tsx [hunk 2]
```diff
@@ -9,8 +9,10 @@ export enum Mode {
 }
 
 export interface InputProps<T> {
```
Comment (pdolega) [inline src/base/Input.tsx]: this `Mode` in general base props doesn't sound as that good idea in the hindsight

File: sample/SamplePage.tsx [hunk 3]
```diff
@@ -36,19 +32,22 @@ class SamplePage extends React.Component<void, State> {
...
             default:
                 return <SamplePageTextInput/>;
         }
     }
 
-    private switchTo(page: Page) {
-        this.state.page = page;
-        this.setState(this.state);
+    private switchToHash() {
+        setInterval(() => {
```
Comment (zagorski) [inline sample/SamplePage.tsx]: You have to use e.preventDefault(). If not, you need to click 2 times to change page.

**Comment by zagorski** [comment]
Besides minor remark :+1: LGTM

File: sample/SamplePage.tsx (hunk 3 above)
Comment (pdolega) [inline sample/SamplePage.tsx]: I just fixed it and will push it on next PR.

**Pull request merged**